### PR TITLE
Improve documentation of kube-prometheus

### PR DESCRIPTION
**Pull request by aknuds1**
Generally improve the kube-prometheus README, recommend creating Kubernetes cluster with Tectonic Installer. Also bring the development documentation up to date, reflecting our usage of grafanalib.

### CODE REVIEW

**Comment by brancz** [comment]
Nice this sounds great. Something that was a frequent point of friction lately is the generation of the dashboards through the `make generate` command. Relatively early we should have a paragraph that the `make generate` command will re-generate all manifests, and that other for deployment the `hack/` scripts should really never be touched. And even when using the `hack/` directory for deployment the name of the directory very much describes the quality.

File: contrib/kube-prometheus/README.md [hunk 1]
```diff
@@ -1,35 +1,40 @@
...
 
-First, you need a running Kubernetes cluster. If you don't have one, follow the
-instructions of [bootkube](https://github.com/kubernetes-incubator/bootkube) or
-[minikube](https://github.com/kubernetes/minikube). Some sample contents of this
+First, you need a running Kubernetes cluster. If you don't have one, we recommend you create one
+with [Tectonic Installer](https://coreos.com/tectonic/docs/latest/). Despite the name,
+Tectonic Installer gives you also the choice to create a barebones Kubernetes cluster, without
+CoreOS' Tectonic technology. Otherwise, you can simply make use of
+[bootkube](https://github.com/kubernetes-incubator/bootkube) or
+[minikube](https://github.com/kubernetes/minikube) for local testing. Some sample contents of this
```
Comment (brancz) [inline contrib/kube-prometheus/README.md]: maybe we can comment that other methods of cluster creation might need a little adaptation, but we're happy to accepts additional docs/PRs to add more compatibility

**Comment by brancz** [comment]
I'll merge this as it's a lot better than what we had before, and my nit was also present previously. Feel free to create another PR to fix that as well, but as for the goal of this PR this is great! Thanks a lot! lgtm :+1:

**Comment by aknuds1** [comment]
Thanks @brancz! :)

**Pull request merged**